tests/unit/test_issues.py

```diff
@@ -35,47 +36,50 @@ def tearDown(self):
     @patch.object(requests, 'post')
     def test_report_issue_returns_number(self, mockpost):
         """Test we can expect an issue number back."""
-        mockpost.return_value.status_code = 201
-        mockpost.return_value.json.return_value = {'number': 2}
-        form = MultiDict([
-            ('browser', 'Firefox 99.0'),
-            ('description', 'sup'),
-            ('details', ''),
-            ('os', 'Mac OS X 13'),
-            ('problem_category', 'unknown_bug'),
-            ('submit_type', 'github-proxy-report'),
-            ('url', 'http://3139.example.com'),
-            ('username', ''), ])
-        rv = report_issue(form, True)
-        self.assertEqual(rv.get('number'), 2)
+        with app.test_request_context():
```

Comment:
What makes the new context `app.test_request_context()` required? Just curious.

Reply:
So I *think* (but am not 100% certain because the universe is a big place) the reason is because I moved the `@mockable_response` decorator to `proxy_request`, which `report_public_issue`, `report_private_issue`, and `report_issue` all eventually call.

Inside of that decorator, we inspect the `request` object: https://github.com/webcompat/webcompat.com/blob/ad693004dcc017fc56b7a2e85f287948db0075a2/webcompat/helpers.py#L401-L402

And in order to look at a `request` you have to have a request context available: https://flask.palletsprojects.com/en/1.1.x/reqcontext/